content/docs/start/experiments/building-pipelines.md

```diff
@@ -0,0 +1,184 @@
+[migrate from notebooks to scripts](https://towardsdatascience.com/from-jupyter-notebook-to-sc-582978d3c0c),
+DVC <abbr>Pipelines</abbr> can help you standardize your workflow following
+software engineering best practices:
+
+- **Modularization**: split the different logical steps in your notebook into
+  separate scripts.
+
+- **Parametrization**: adapt your scripts to decouple the configuration from the
+  source code.
+
+## Creating stages
+
+In our example repo, we first extract data preparation from the
+[original notebook](https://github.com/iterative/example-get-started-experiments/blob/main/notebooks/TrainSegModel.ipynb)
+into
+[`data_split.py`](https://github.com/iterative/example-get-started-experiments/blob/main/src/data_split.py).
+We parametrize this script by reading parameters from
+[`params.yaml`](https://github.com/iterative/example-get-started-experiments/blob/main/params.yaml):
+
+```python
+from ruamel.yaml import YAML
+
+yaml = YAML(typ="safe")
+
+def data_split():
+    params = yaml.load(open("params.yaml", encoding="utf-8"))
+...
+```
+
+You can use `dvc stage add` to transform a script into a <abbr>stage</abbr>:
+
+```cli
+$ dvc stage add --name data_split \
+  --params base,data_split \
+  --deps data/pool_data --deps src/data_split.py \
+  --outs data/train_data --outs data/test_data \
+  python src/data_split.py
+```
+
+A `dvc.yaml` file is generated. It includes information about the command you
+want to run (`python src/data_split.py`), its <abbr>dependencies</abbr>,
+<abbr>parameters</abbr>, and <abbr>outputs</abbr>:
+
+```yaml
+stages:
+  prepare:
+    cmd: python src/data_split.py
+    deps:
+      - src/data_split.py
+      - data/pool_data
+    params:
+      - base
+      - data_split
+    outs:
+      - data/train_data
+      - data/test_data
+```
+
+`dvc exp run` will run all stages in the `dvc.yaml` file.
+
+<admon type="info">
+
+Learn more about [Stages](/doc/user-guide/pipelines/defining-pipelines#stages)
+
+</admon>
+
+## Building a DAG
+
+By using `dvc stage add` multiple times and defining <abbr>outputs</abbr> of a
+stage as <abbr>dependencies</abbr> of another, you describe a sequence of
+commands which forms a
+[pipeline](https://dvc.org/doc/user-guide/pipelines/defining-pipelines), also
+called a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph).
+
+Let's create a `train` stage using
+[`train.py`](https://github.com/iterative/example-get-started-experiments/blob/main/src/train.py)
+to train the model:
+
+```cli
+$ dvc stage add -n train \
+  -p base,train \
+  -d src/train.py -d data/train_data \
+  -o models/model.pkl \
+  python src/train.py
+```
+
+`dvc exp run` checks the `prepare` stage first and then the `train` stage since
+it depends on the <abbr>outputs</abbr> of `prepare`. If you already ran
+`prepare` with the same <abbr>dependencies</abbr> and <abbr>parameters</abbr>,
+it will be retrieved from the [run cache](/doc/user-guide/pipelines/run-cache)
+and skipped.
```

Comment:
> Not blocking:
> 
> Do you think is worth differentiating between retrieving from the cache and skipping because nothing changed?
> Maybe is obvious but we don't mention the skipping if nothing changed

_Originally posted by @daavoo in https://github.com/iterative/dvc.org/pull/4425#discussion_r1147313995_

Reply:
I originally mentioned skipping because nothing changes, but then decided to copy your text from the next page to link to the run cache. I don't think we should explain both/differentiate in the get started, although maybe in the guide it's worth it. I think it's probably better to make it general here. WDYT?

Reply:
> lthough maybe in the guide it's worth it. I think it's probably better to make it general here. WDYT?

Sounds good. 
Let's remember to cover it in the guide.